12.0 split beesdoo_base

victor-champonnois opened the pull request:
fix https://github.com/beescoop/Obeesdoo/issues/335
fix https://github.com/beescoop/Obeesdoo/issues/334
fix https://github.com/beescoop/Obeesdoo/issues/336

Split beesdoo_base into eater, member_card and eater_member_card (code + migration)

Having separate PR for each issue is really painful to maintain, so I prefered making one PR. 
related PR in addons https://github.com/coopiteasy/addons/pull/244

Still todo : 
- correct the translations
- ~~there are still some small functionalities in beesdoo_base, some might be obsolete (portal wizard stuff), and some might be put in newly refactored modules (see after beesdoo_easy_my_coop refactoring)~~
FIXED : removed or added to addons : https://github.com/coopiteasy/addons/pull/244
- related error : https://github.com/coopiteasy/vertical-cooperative/issues/340

Note : this PR conflicts with https://github.com/beescoop/Obeesdoo/pull/289

robinkeunen commented:
Ok to move ahead regardless of conflicts w/ #289

robinkeunen commented:
Hard to review, I tend to trust automatic and functionnal tests on this one.

robinkeunen commented on beesdoo_base/views/partner.xml:
```diff
@@ -1,98 +1,13 @@
-            </field>
             <!-- Temporary fix for the partner_view -->
             <xpath expr="//field[@name='customer']" position='before'>
                 <field name="is_company" invisible="True" />
```
can't comment directly on below code...

```py
<record id="view_beesdoo_res_partner_filter" model="ir.ui.view">
```
=> move to `member_card`

```py
<record id="base.action_partner_supplier_form" model="ir.actions.act_window">
```
 => should go in a separate module in `coopiteasy/addons`, it won't be accepted directly in OCB

victor-champonnois replied:
done : https://github.com/coopiteasy/addons/pull/244

robinkeunen commented on beesdoo_base/views/partner.xml:
```diff
@@ -1,98 +1,13 @@
-                />
-            </field>
             <!-- Temporary fix for the partner_view -->
             <xpath expr="//field[@name='customer']" position='before'>
```
I'm pretty sure this can be removed : the change dates from 2019 (9.0) and in base view, I see the field `is_company` was added in 2020 before the field `customer`.

victor-champonnois replied:
done

robinkeunen commented on beesdoo_base/wizard/__init__.py:
```diff
@@ -1,3 +1 @@
-from . import member_card
-from . import partner
 from . import portal_wizard
```
These are ancestral changes https://github.com/beescoop/Obeesdoo/commit/7c6a7ff4ac3191813b3aaa222949a4c8aa430024, I think they could be removed after a little investigation.

victor-champonnois replied:
changes removed

robinkeunen reviewed (changes requested):
Could not review everything, I'm happy if the unit tests pass and a few functional tests. I suggested changes for the leftovers in `views` and `wizard`

victor-champonnois commented:
- migration during update is OK. Although it only works when updating only beesdoo_base, not an update all (it seems it cannot find the company_supplier_context dependency).
Only issue is no uninstall of the empty beesdoo_base module.

robinkeunen commented on member_card/__init__.py:
```diff
@@ -1,2 +1,52 @@
 from . import models
 from . import wizard
+
+
+from openupgradelib import openupgrade
+from odoo import api, SUPERUSER_ID
+
+
+renamed_view_xml_ids = (
+    (
```
All of this should rather be put in a file `pre_init_hook.py` and then you import that file in this `__init__.py` file. Check out the examples in the code.

victor-champonnois replied:
done

robinkeunen reviewed (changes requested):
Un commentaire de rangement, otherwise lgtm

victor-champonnois commented:
/ocabot rebase

victor-champonnois commented:
@vdewulf Ready to test on lenid-test ! It must be ensured that all the beesdoo_base functionalities are still present: 
- Add eater to a partner (`eater` module)
- Create a member card and link it to a partner (`member_card` module)
- Compute barcode based on eaters (`eater_member_card` module)
- Keep only companies in supplier context (`company_supplier_context` module)

Note : there is a minor functional change: in order to decouple the modules `eater` and `member_card`, there is now a tab "Eater" and a tab "Member Card" in the Contact form.

victor-champonnois commented:
The CI failing test is related to beesdoo_shift_swap

victor-champonnois commented:
@robinkeunen I removed the field `cooperator_type` from beesdoo_base, and I did not put it in any splitted module so far because it did not fit the description. The field was defined as follow :
```
cooperator_type = fields.Selection(
      [
          ("share_a", "Share A"),
          ("share_b", "Share B"),
          ("share_c", "Share C"),
      ],
      store=True,
      compute=None,
  )
```
It looks very "hardcoded" to me. Do you know why is it coded like this ? 
I am considering to put it in the new `cooperator_worker_configuration`, but I am not sure, do you have a better idea ? Maybe member_card ?

victor-champonnois commented:
/ocabot rebase

vdewulf commented:
@victor-champonnois Here are my test results

SUMMARY
- Add eater to a partner (eater module) ==> tested, not ok
- Create a member card and link it to a partner (member_card module) ==> tested, seems ok
- Compute barcode based on eaters (eater_member_card module) ==> I don't get what this is (but I tested member card tab on an eater contact, there are some issues)
- Keep only companies in supplier context (company_supplier_context module) ==> I don't get what this is. I don't see the related code into beesdoo_base module.

DETAILS
Testing beesdoo_base split on lenid.test.coopiteasy.be

- Creation of a user who has only the right to use cooperative management and accounting (facturation)
- login with this user
- go to cooperators menu and validate a draft request (to male the whole process of creating a new coop and his eaters). I've got this error message:
" Désolé, vous n’êtes pas autorisé à modifier ce document. Seuls les utilisateurs avec les niveaux d'accès suivants sont actuellement autorisés à faire cela: - Administration/Access Rights"
==> not linked to this PR but maybe linked to cooperator refactor?
- to go on with the test, I add this user rights to my user
- I retry, this error occurs:
Désolé, vous n'êtes pas autorisé à créer ce type de document. Seuls les utilisateurs avec les niveaux d'accès suivants sont actuellement autorisés à faire cela: - Other Extra Rights/Accounting / Payments
- I add this user rights again (to be checked, though)
- I validate the request (Hello WORLD) (but don't register payment, so this person is coop candidate)
- the contact is created and has the Member Card tab and Eaters tab. Eater/Worker = Worker and Eater as expected.
- on the Member card tab, I click on "New Card".
- pop up opens with "motif". I validate and the error is displayed:
L'opération demandée ne peut pas être effectuée en raison de la politique de sécurité. Veuillez contacter votre administrateur système.
(Type de document : Member Card, Opération : read) - (Utilisateur : 1560, Champs: force_barcode) 
- I add this user right (I think this is already needed before refactor, so not an issue)
- I create the new card. Barcode is generated (042...), date of creation is set, responsible as well (my user) and the motif. The card is activated. I create a new one. New line is added and is active, the first line is no inactive.
Print BEES card? is set.
- I tested the 2 actions "Request BEES card printing" and "Set BEES card as printed". It behaves as expected (Print BEEScard? and Last Printed on are updated).
- On tab "Eaters", I add a new Eater. I was able to create 3 eaters.I checked the config of "part sociale" and they may have 9 eaters. I tried to change this config but was not allowed because of user right issue:
Désolé, vous n'êtes pas autorisé à accéder à ce document. Seuls les utilisateurs avec les niveaux d'accès suivants sont actuellement autorisés à faire cela:
- Administration/Settings
(Modèle de document: ir.config_parameter) - (Opération: read, Utilisateur : 1560) 
==> Is it normal?

- I added more than 10 Eaters and got no error validation.
- Another issue: before refactoring (I suppose), it was possible to search for an existing contact to add an Eater on a Worker contact. Now this field is not available anymore.

- On the Eater contact, there is the Eater/Worker = Eater and the Parent Eater is well displayed (not editable)
On the Member Card tab, there is Print BEES card? (set to true by default after creation of the eater) and Last printed on (empty). There is also the table "Cards" but this is not displayed on Le nid production database, so I suppose is not ok? Also the "New card" button is displayed and should not.

vdewulf commented:
Following meetint virg-victor

- go to cooperators menu and validate a draft request (to male the whole process of creating a new coop and his eaters). I've got this error message:
" Désolé, vous n’êtes pas autorisé à modifier ce document. Seuls les utilisateurs avec les niveaux d'accès suivants sont actuellement autorisés à faire cela: - Administration/Access Rights"
==> not linked to this PR but maybe linked to cooperator refactor?
TODO Virg : test on a database with cooperator and without beesdoo_base (brewery for example)
Create issue if it's the case 'on cooperator)

Victor: fixes issues ;)

vdewulf commented:
@victor-champonnois 
I tested on brasseriedurenard.test.coopiteasy.be (only Cooperators, no beesdoo code) the validation of a subscription request for a future cooperator.
The user I was logged in had the following right:
- Cooperative Management : User
- Accounting : Facturation
- Creation de contact
I was able to validate the request without error.

So I cannot reproduce the error I got on lenid (" Désolé, vous n’êtes pas autorisé à modifier ce document. Seuls les utilisateurs avec les niveaux d'accès suivants sont actuellement autorisés à faire cela: - Administration/Access Rights")
This seems that the error might be linked to the refacto of beesdoo_base here.

I'll let you try and reproduce it, and hopefully fix it, right?
Thanks

victor-champonnois commented:
@vdewulf Thank you for your test. I think it is related to cooperator then. I create a specific issue in vertical cooperative https://github.com/coopiteasy/vertical-cooperative/issues/340

victor-champonnois commented:
@vdewulf 
- regarding the permission issues, I think it is not related to beesdoo_base. I created an issue in vertical-cooperative. Also I think that the process you were trying to test (validating a draft request) is related to `beesdoo_easy_my_coop`, not `beesdoo_base`. 
- regarding the ability to search eaters when adding eaters, I cannot reproduce your bug. The contact completion works fine for me in lenid-test.
- regarding the issue with eaters that could see the barcode table and see the New card button, it is fixed now. 
- regarding the ability to create more than 9 eaters, this is related to beesdoo_easy_my_coop, not beesdoo_base. I created a new issue.  https://github.com/beescoop/Obeesdoo/issues/429
- I also renamed BEES card into Member card.

Ready to be retested on lenid-test

vdewulf commented:
@victor-champonnois 

SUMMARY
Everything seems good except when trying to add an existing contact as "eater" from the worker contact.

DETAILS
Testing beesdoo_base split on lenid.test.coopiteasy.be
- Creation of a user who has only the right to use cooperative management and accounting (facturation) + member card force barcode
- login with this user
-  go to cooperators menu and select a cooperator (worker and eater). He has no member card.
- on the Member card tab, I click on "New Card". Everything ok as expected
- On tab "Eaters",  it is possible again to search for an existing contact to add an Eater on a Worker contact. I try to add someone, I get this error message:


Odoo Server Error

Traceback (most recent call last):
  File "/home/odoo12/src/odoo/odoo/http.py", line 656, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo12/src/odoo/odoo/http.py", line 314, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/odoo12/src/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/odoo12/src/odoo/odoo/http.py", line 698, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo12/src/odoo/odoo/http.py", line 346, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo12/src/odoo/odoo/service/model.py", line 98, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo12/src/odoo/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo12/src/odoo/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "/home/odoo12/src/odoo/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo12/src/odoo/addons/web/controllers/main.py", line 967, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo12/src/odoo/addons/web/controllers/main.py", line 959, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo12/src/odoo/odoo/api.py", line 759, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/home/odoo12/src/odoo/odoo/api.py", line 746, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo12/src/odoo/odoo/models.py", line 5527, in onchange
    record = self.new(values)
  File "/home/odoo12/src/odoo/odoo/models.py", line 4989, in new
    record._cache.update(record._convert_to_cache(values, update=True))
  File "/home/odoo12/src/odoo/odoo/models.py", line 4859, in _convert_to_cache
    for name, value in values.items()
  File "/home/odoo12/src/odoo/odoo/models.py", line 4860, in <dictcomp>
    if name in fields
  File "/home/odoo12/src/odoo/odoo/fields.py", line 2064, in convert_to_cache
    raise ValueError("Wrong value for %s: %r" % (self, value))
ValueError: Wrong value for res.partner.cooperator_type: 'part sociale'


- I click on "add a new Eater" button. I has no user right error message. I tried to add more than 10 Eaters and got the error message " You can only set 9 additional eaters per worker "
- On the Eater contact, on the Member Card tab, there is Print BEES card? (set to true by default after creation of the eater) and Last printed on (empty). There is no more the table "Cards" neither the "New card" button.

- Tab "Member Card" is well displayed (instead of BEES card, as well as in the Action menus) but the "Print BEES card?" checkbox uses the old name. I suppose it's a matter of translation to update in the production database if needed, and I suppose that for a new installation from scratch, the word will be correctly set to Member Card

victor-champonnois commented:
Discussion with robin regarding the cooperator_type variable

Me
>The field cooperator_type is defined [in cooperator](https://github.com/coopiteasy/vertical-cooperative/blob/35443a5179fff137b1f7e69a09cc7b9b17412343/cooperator/models/partner.py#L159), but also defined in a very ad hoc manner [in beesdoo_base](https://github.com/beescoop/Obeesdoo/blob/449a040c467b9bfe72476f3e57ecd0f63a881b04/beesdoo_base/models/partner.py#L34). I don't understand how it didn't break so far. Can the second one be safely removed ? 

Robin
>It's :shit: but it does not break anything. It just prevents any supermarket to have other shares than A, B, C. Before removing this, you'd need to check 1) if all supermarkets have their shares properly setup and 2) if those share string are used in the beesdoo_code
=> many demo data to fix
=> some views use the share type
=> some python searches uses it
=> ...
not that trivial

me:
conclusion : don't remove cooperator_type but put it in beesdoo_easy_my_coop.

victor-champonnois commented:
related to this issue https://github.com/coopiteasy/vertical-cooperative/issues/345

victor-champonnois committed:
[REM] deprecated functionalities

[FIX] contributors

[FIX] remove deprecated demo data

victor-champonnois committed:
[REF] split beesdoo_base -> member_card

[UPD] Update member_card.pot

victor-champonnois committed:
[REF] split beesdoo_base : fix dependencies

[REF] put splitted modules as dependencies

[FIX] test-requirements

[FIX] remove beesdoo_base demo data in manifest

victor-champonnois committed:
[REF] split beesdoo_base -> eater

[UPD] Update eater.pot

victor-champonnois committed:
[REF] split beesdoo_base -> eater_member_card

[UPD] Update eater_member_card.pot

victor-champonnois committed:
[REF] move cooperator_type field to beesdoo_easy_my_coop

victor-champonnois committed:
[FIX] reference to member_card demo

victor-champonnois commented:
>Tab "Member Card" is well displayed (instead of BEES card, as well as in the Action menus) but the "Print BEES card?" checkbox uses the old name. I suppose it's a matter of translation to update in the production database if needed, and I suppose that for a new installation from scratch, the word will be correctly set to Member Card

Indeed it works well in a fresh DB.

>On tab "Eaters", it is possible again to search for an existing contact to add an Eater on a Worker contact. I try to add someone, I get this error message:
Odoo Server Error
Traceback (most recent call last):
File "/home/odoo12/src/odoo/odoo/http.py", line 656, in _handle_exception
return super(JsonRequest, self)._handle_exception(exception)
File "/home/odoo12/src/odoo/odoo/http.py", line 314, in _handle_exception
raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
File "/home/odoo12/src/odoo/odoo/tools/pycompat.py", line 87, in reraise
raise value
File "/home/odoo12/src/odoo/odoo/http.py", line 698, in dispatch
result = self._call_function(**self.params)
File "/home/odoo12/src/odoo/odoo/http.py", line 346, in _call_function
return checked_call(self.db, *args, **kwargs)
File "/home/odoo12/src/odoo/odoo/service/model.py", line 98, in wrapper
return f(dbname, *args, **kwargs)
File "/home/odoo12/src/odoo/odoo/http.py", line 339, in checked_call
result = self.endpoint(*a, **kw)
File "/home/odoo12/src/odoo/odoo/http.py", line 941, in call
return self.method(*args, **kw)
File "/home/odoo12/src/odoo/odoo/http.py", line 519, in response_wrap
response = f(*args, **kw)
File "/home/odoo12/src/odoo/addons/web/controllers/main.py", line 967, in call_kw
return self._call_kw(model, method, args, kwargs)
File "/home/odoo12/src/odoo/addons/web/controllers/main.py", line 959, in _call_kw
return call_kw(request.env[model], method, args, kwargs)
File "/home/odoo12/src/odoo/odoo/api.py", line 759, in call_kw
return _call_kw_multi(method, model, args, kwargs)
File "/home/odoo12/src/odoo/odoo/api.py", line 746, in _call_kw_multi
result = method(recs, *args, **kwargs)
File "/home/odoo12/src/odoo/odoo/models.py", line 5527, in onchange
record = self.new(values)
File "/home/odoo12/src/odoo/odoo/models.py", line 4989, in new
record._cache.update(record._convert_to_cache(values, update=True))
File "/home/odoo12/src/odoo/odoo/models.py", line 4859, in _convert_to_cache
for name, value in values.items()
File "/home/odoo12/src/odoo/odoo/models.py", line 4860, in
if name in fields
File "/home/odoo12/src/odoo/odoo/fields.py", line 2064, in convert_to_cache
raise ValueError("Wrong value for %s: %r" % (self, value))
ValueError: Wrong value for res.partner.cooperator_type: 'part sociale'

-> To fix
Note, this could be related to the fact that lenid does not have the share configuration A/B/C. It only have one share "part sociale".
To do:
- download lenid DB locally and debug the widget 
```
                          <field
                                name="child_eater_ids"
                                string="Eaters"
                                widget="many2many_tags"
                                options="{'no_create': True}"
                                context="{'default_eater' : 'eater', 'default_customer' : True}"
                            />
```

robinkeunen committed:
[REF] beesdoo_base: remove view renames

We do not need to rename view names when moved from one
module to the other. Odoo cleans up by himself as long
as they are not defined as noupdate.

robinkeunen commented:
@vdewulf @polchampion This development has been cleaned up and deployed on these databases : 

```
bablmarket-test
bees-test
bloum-test
coopeco-test
demain-test
fhmunich-test
gastropotes-test
lasource-test
lenid-test
lepedalo-test
lesptitspots-test
oufticoop-test
shopbrussels-test
shopleuven-test
spp-test
vervicoop-test
wecostore-test
wholesale-test
```

BUT, these databases have shares' default_code inconsistent with the cooperator_type defined in beesdoo_easy_my_coop ;

- bees-test (but mostly fixed) - fit for tests \*
- demain-test - unfit for tests
- fhmunich-test - unfit for tests
- lasource-test - unfit for tests
- lenid-test - unfit for tests
- lepedalo-test - unfit for tests
- vervicoop-test - unfit for tests

The misconfiguration leads to the error you raised earlier (setting and removing eaters on a partner). So I suggest you test this split of beesdoo_base on other database than the last 6.

After validating this, I can work on beesdoo_easy_my_coop and refactor our code so that it can handle better the cooperator_types.

To test this, you should check that this modules are installed

- eater
- member_card
- eater_member_card
- company_supplier_context

and cover these use cases this use cases 

- Add eater to a partner (eater module)
- Create a member card and link it to a partner (member_card module)
- Compute barcode based on eaters (eater_member_card module)
- When creating a new supplier, make it a company partner type by default (company_supplier_context module)

\* avoid using the 20 partners who have shares "Abis"

\*\* I won't make a pun about the Abyss the misconfiguration sent us in.

vdewulf commented:
@robinkeunen (@polchampion fyi)
Tested on bees.test.coopiteasy.be and bloum.test today.

EATER module
* eater: the tab "eater" on partner is well displayed for the cooperators I checked. I added new eaters, OK. I added a 3th eater on someone with share A type, I got an error message (On ne peut ajouter que 2 mangeurs par travailleur)
* I changed the "max number of eater allowed" on share A to 3 and added a 3th again. Works OK.
* [QUESTION] I tested also on lesptitspots because they don't have cooperator (to check the behavior in this context). There is a field "travailleur", I set it to true and the eater tab was displayed (+ eater/worker = worker and eater). I could add 4 workers. I'm wondering if there is a config somewhere else than the shares to configure a max (I don't know how it was before refactor).
* [TODO] the readme could be improved I think "add eater to a partner". I would add the context (what is a eater, how to display the tab "eater" on a partner, how you can configure the max eater allowed if other module installed).

MEMBER CARD
*[TODO] readme could be improved as well (explain that there is no member card to be printed, only a barcode to be generated and history of the barcode generated through time).
* [TODO] I clicked on "new card" button on Member card tab on a partner (worker and eater), I added a reason and clicked on Create. I got an error linked to user right: L'opération demandée ne peut pas être effectuée en raison de la politique de sécurité. Veuillez contacter votre administrateur système.
(Type de document : Member Card, Opération : read) - (Utilisateur : 330, Champs: force_barcode) 
This is an old behavior but I still don't understand this need or this name of user right "force barcode". I remember it should mean that the user can write the barcode numbers himself instead of having the system generate it. But in reality you need to have this user right even to create a new barcode. I added this user right on admin and on the pop up to indicate the reason, there is the "force barcode" tab available as well.
I propose to either remove this user right totally either keep it but allow people to create a member card (the only restriction if you don't have this user right is to encode something on the force barcode field).
* I was then able to create new card, force the barcode, or not. The generated barcode is 042..., it's ok. The active and expiration date mechanism is OK.
* "imprimer la carte ?" is set to true automatically. I clicked on action > marquer la carte comme imprimée > "imprimer la carte" is set to false and "dernière impression" is updated with today time.
* [TODO] I got this user right error message on several databases, so the migration script should keep this value on the existing data
* [TODO] I clicked on "Imprimer" > Member card and I got a 500 error messages. Should we remove this print button if the code is not working now?

EATER MEMBER CARD // eater_member_card
*[TODO] readme could be improved as well (I don't see where there is a "computation" of the eater barcode here).
- on an eater partner, the tab "carte de membre" is displayed. There are only 2 fields: "imprimer la carte ?" and "dernière impression".
- [TO CHECK] as there is no barcode there, I exported the eater and the corresponding worker with the member card fields. There is no member card info on the eater. I suppose it's normal because it's managed by using the one from the worker? It's the same on production database.
- I clicked on Action > "demander l'impression de la carte" then on "marquer la carte comme imprimée", it works fine.
- [QUESTION] I don't know what I should test more

COMPANY SUPPLIER CONTEXT // company_supplier_context
- I created a new supplier, the "company" type is automatically selected. OK

-----------------------------------
I'm thinking that Bloum might not need these new modules (maybe we installed them because of dependencies). They don't have people working in the shop and I don't think they provide member cards. I'll check with Matthieu.

robinkeunen committed:
[REF] eater: rename eater.new.wizard

robinkeunen committed:
[REF] eater: dependencies to point_of_sale, purchase

robinkeunen committed:
[REF] member_card: rename membercard.new.wizard

robinkeunen committed:
[IMP] member_card: remove group_force_barcode

robinkeunen commented:
Merci pour le super feedback :clap: @vdewulf @polchampion 

I deployed the following modifications on bees-test and bloum-test.

I still need to cleanup translations.

## EATER module

> [QUESTION] I tested also on lesptitspots because they don't have cooperator (to check the behavior in this context). There is a field "travailleur", I set it to true and the eater tab was displayed (+ eater/worker = worker and eater). I could add 4 workers. I'm wondering if there is a config somewhere else than the shares to configure a max (I don't know how it was before refactor).

There are no configuration about the number of eater in this modules. It comes in `cooperator_eater_configuration` in the refactor of [beesdoo_easy_my_coop](https://github.com/beescoop/Obeesdoo/pull/427/files) _not deployed, not ready_.

It is indeed defined on the share type but I wonder if it should rather be a company parameter and brought back in this module.

 
> [TODO] the readme could be improved I think "add eater to a partner". I would add the context (what is a eater, how to display the tab "eater" on a partner, how you can configure the max eater allowed if other module installed).

DONE

### Other improvements

- removed the dependencies to point_of_sale and purchase.
- `eater.new.wizard` renamed to `new.eater.wizard`. Translations and views were adapted (but translations need a cleanup).

=> The module still installs ok and the tests do pass. 

## MEMBER CARD

> [TODO] readme could be improved as well (explain that there is no member card to be printed, only a barcode to be generated and history of the barcode generated through time).

DONE

> [TODO] I clicked on "new card" button on Member card tab on a partner (worker and eater), I added a reason and clicked on Create. I got an error linked to user right: L'opération demandée ne peut pas être effectuée en raison de la politique de sécurité. Veuillez contacter votre administrateur système.
> (Type de document : Member Card, Opération : read) - (Utilisateur : 330, Champs: force_barcode)
> This is an old behavior but I still don't understand this need or this name of user right "force barcode". I remember it should mean that the user can write the barcode numbers himself instead of having the system generate it. But in reality you need to have this user right even to create a new barcode. I added this user right on admin and on the pop up to indicate the reason, there is the "force barcode" tab available as well.
> I propose to either remove this user right totally either keep it but allow people to create a member card (the only restriction if you don't have this user right is to encode something on the force barcode field).

DONE: removed the security group altogether

> [TODO] I got this user right error message on several databases, so the migration script should keep this value on the existing data

I don't undestand this remark. What should we keep ?

> [TODO] I clicked on "Imprimer" > Member card and I got a 500 error messages. Should we remove this print button if the code is not working now?

I removed this report since it's broken and contains many references to the BEES coop. I added a link to the roadmap to Thibault's PR for Polln Card.

### Other improvements

- removed dependencies to point_of_sale, purchase, portal
- the generated barcode follows the customer pattern only if point_of_sale is installed
- renamed membercard.new.wizard to new.member.card.wizard. Translations and views were adapted (but translations need a cleanup).

## EATER MEMBER CARD

> [TODO] readme could be improved as well (I don't see where there is a "computation" of the eater barcode here).

DONE

> on an eater partner, the tab "carte de membre" is displayed. There are only 2 fields: "imprimer la carte ?" and "dernière impression".
> [TO CHECK] as there is no barcode there, I exported the eater and the corresponding worker with the member card fields. There is no member card info on the eater. I suppose it's normal because it's managed by using the one from the worker? It's the same on production database.

That's it, as I added to the DESCRIPTION.rst, the eater barcode is computed from parent's barcode.

> I clicked on Action > "demander l'impression de la carte" then on "marquer la carte comme imprimée", it works fine.
> [QUESTION] I don't know what I should test more

That's it :+1:

### Other improvements

- removed dependencies to point_of_sale, purchase
- when new eater was created, `member_card_to_be_printed` was not correctly set to True. Fixed it.

robinkeunen committed:
[IMP] generate POT and PO files

robinkeunen committed:
[IMP] update french translations

robinkeunen commented:
@vdewulf Peux-tu réviser les traductions sur ce commit ? https://github.com/beescoop/Obeesdoo/pull/384/commits/e92f36784f55c439c1d092a898cedac17170ce91

vdewulf commented:
EATER module : 
- I don't see the improved readme on the app list neither in the code here on Github
- I tested the encoding of a new eater on someone, the limit max is still working, it's OK.

MEMBER CARD
- I don't see the improved readme on the app list neither in the code here on Github
- User group is removed (I don't see it on admin user). I could create a new member card for a user. The reason and force barcode are both displayed. Both functionalities are working fine.
- Robin wrote "I don't undestand this remark. What should we keep ?" => it was linked to the user group, you removed it so my comment is obsolete.
- Action button "imprimer" is removed from the contact form (form view) and from the list view, the print button allows to print certificate coop things, it's ok.

EATER MEMBER CARD
- I don't see the improved readme on the app list neither in the code here on Github
- I created an eater from a worker contact view, then checked the eater form view. The tab "carte de membre" has the info "imprimer la carte ?" set to true by default.

vdewulf commented:
> @vdewulf Peux-tu réviser les traductions sur ce commit ? [e92f367](https://github.com/beescoop/Obeesdoo/commit/e92f36784f55c439c1d092a898cedac17170ce91)

Done

robinkeunen commented:
@vdewulf the README.rst are generated at merge time by the OCABOT => have a look in the DESCRIPION.rst files

To save you some time :

## eater
This module adds the "eater" and "worker" concepts to partners of your participative shop.

- the worker is a voluntary worker of your supermarket
- the eater, linked to a worker, is usually given the same access rights as the worker

The eater tab is displayed for worker partner.
A button in this tab allows to create and link an eater to a partner.

## member_card

Create a member card and link it to a partner.

- each member card has a barcode
- create new cards in the "member card" tab of the partner form
- use "Force Barcode" field to set a user defined barcode
- the partner's card and barcode history is visible in the member card tab
- the barcode of a partner is computed from the last active member card
- a boolean "Print Member card?" allows to filter on partners for which you need to print new cards.

Careful : this module overrides the barcodes already defined on the partners.

Note that there is no member card template in this module, cf ROADMAP.

If point of sale is installed, the generated barcode matches customer pattern rule.
 
## eater_member_card

- eater barcode is computed from parent's barcode.
- Print Member card? is set to True when a new eater is created.

vdewulf commented:
@robinkeunen perfect for me, then :) thanks for the copy/paste and my time saving

robinkeunen commented:
/ocabot merge nobump

robinkeunen commented:
/ocabot merge nobump
